[Console] Errors on console

Observing repeated errors on console during testathon 20160808:

```
main.js:3 Error: [$rootScope:inprog] http://errors.angularjs.org/1.4.4/$rootScope/inprog?p0=%24digest
    at Error (native)
    at https://hulk.arc.nasa.gov:8080/main.js:1:21188
    at f (https://hulk.arc.nasa.gov:8080/main.js:3:17432)
    at p.$apply (https://hulk.arc.nasa.gov:8080/main.js:3:21481)
    at https://hulk.arc.nasa.gov:8080/main.js:14:23017
    at https://hulk.arc.nasa.gov:8080/main.js:14:22622
    at Array.forEach (native)
    at i.value (https://hulk.arc.nasa.gov:8080/main.js:14:22602)
    at p.$digest (https://hulk.arc.nasa.gov:8080/main.js:3:20049)
    at p.$apply (https://hulk.arc.nasa.gov:8080/main.js:3:21570)
```

Haven't determined what triggers these errors yet.

--- a/platform/commonUI/general/src/directives/MCTTree.js
+++ b/platform/commonUI/general/src/directives/MCTTree.js
@@ -28,8 +28,10 @@ define([
         function link(scope, element) {
             var treeView = new TreeView(gestureService),
                 unobserve = treeView.observe(function (domainObject) {
-                    scope.mctModel = domainObject;
-                    scope.$apply();
+                    if (scope.mctModel !== domainObject) {
+                        scope.mctModel = domainObject;
+                        scope.$apply();
+                    }
                 });
 
             element.append(angular.element(treeView.elements()));

--- a/platform/commonUI/general/test/directives/MCTTreeSpec.js
+++ b/platform/commonUI/general/test/directives/MCTTreeSpec.js
@@ -29,6 +29,18 @@ define([
             mockExpr,
             mctTree;
 
+        function makeMockDomainObject(id) {
+            var mockDomainObject = jasmine.createSpyObj('domainObject-' + id, [
+                'getId',
+                'getModel',
+                'getCapability',
+                'hasCapability'
+            ]);
+            mockDomainObject.getId.andReturn(id);
+            mockDomainObject.getModel.andReturn({});
+            return mockDomainObject;
+        }
+
         beforeEach(function () {
             mockGestureService = jasmine.createSpyObj(
                 'gestureService',
@@ -56,7 +68,8 @@ define([
                 testAttrs;
 
             beforeEach(function () {
-                mockScope = jasmine.createSpyObj('$scope', ['$watch', '$on']);
+                mockScope =
+                    jasmine.createSpyObj('$scope', ['$watch', '$on', '$apply']);
                 mockElement = jasmine.createSpyObj('element', ['append']);
                 testAttrs = { mctModel: "some-expression" };
                 mockScope.$parent =
@@ -87,6 +100,27 @@ define([
                     "$destroy",
                     jasmine.any(Function)
                 );
+            });
+
+            // https://github.com/nasa/openmct/issues/1114
+            it("does not trigger $apply during $watches", function () {
+                mockScope.mctObject = makeMockDomainObject('root');
+                mockScope.mctMode = makeMockDomainObject('selection');
+                mockScope.$watch.calls.forEach(function (call) {
+                    call.args[1](mockScope[call.args[0]]);
+                });
+                expect(mockScope.$apply).not.toHaveBeenCalled();
+            });
+            it("does trigger $apply from other value changes", function () {
+                // White-boxy; we know this is the setter for the tree's value
+                var treeValueFn = mockScope.$watch.calls[0].args[1];
+
+                mockScope.mctObject = makeMockDomainObject('root');
+                mockScope.mctMode = makeMockDomainObject('selection');
+
+                treeValueFn(makeMockDomainObject('other'));
+
+                expect(mockScope.$apply).toHaveBeenCalled();
             });
         });
     });

